hatchways/team-granite
# Profile Photo and Editing Front End

### What this PR does (required):

- Added 2 new components and 1 page.
- ImageUpload is a component that handles drag and drop of files along with showing a preview of it. Allows for onUpload handler to upload the image to the server.
- EditProfilePhotoDialog is a component that allows for a pop up and has the ImageUpload component within it.
- ChatSideBanner component was modified to allow for the edit profile photo dialog box pop us and a function to modify the profile photo in real time can be passed to the ImageUpload component to allow for real time updating.
- EditProfilePage was added which is a repurposed version of the register page.
- 2 Forms were added , EditProfileForm and ChangePasswordForm.
- Added 4 new API helpers updateProfile, changePassword, uploadProfilePhoto and getImageSource.

### Screenshots / Videos (front-end only):
Edit Profile Form - Repurposed from register, very simple design users can add in the new details in any of the fields and modifies it based on that.
![PR2](https://user-images.githubusercontent.com/55607809/139572493-1866dc31-b462-4451-8ce7-f5157dc759c9.png)

Modify Profile Photo Dialog Box - Allows users to drag and drop the image and have a look at the preview before uploading.
![PR3](https://user-images.githubusercontent.com/55607809/139507922-e391bcd4-d068-4337-90b3-bccaf0a5228e.png)

### Any information needed to test this feature (required):
- register a new user and login
- click on the additional options icon and click on Edit Profile to go to the edit profile page.
- click on the avatar to check the upload image pop-up. 

### Any issues with the current functionality (optional):
- Has no functionality and are place holders for the next issue.
- Typing was a challenge(python experience) so had to disable it on the ImageUpload file as the types were not working well with react dropzone.
- I did take design liberty in moving the upload profile image outside the edit profile settings and on click in the avatar image. I could only allow that dialog box to pop in the edit profile page.
- The api helpers do not function and are placeholders for me to implement it for the next issue.
